packaging/_parser.py

```diff
@@ -0,0 +1,200 @@
+from typing import Any, List, NamedTuple, Tuple, Union
+
+from ._tokenizer import Tokenizer
+
+
+class Node:
+    def __init__(self, value: Any) -> None:
+        self.value = value
+
+    def __str__(self) -> str:
+        return str(self.value)
+
+    def __repr__(self) -> str:
+        return f"<{self.__class__.__name__}('{self}')>"
+
+    def serialize(self) -> str:
+        raise NotImplementedError
+
+
+class Variable(Node):
+    def serialize(self) -> str:
+        return str(self)
+
+
+class Value(Node):
+    def serialize(self) -> str:
+        return f'"{self}"'
+
+
+class Op(Node):
+    def serialize(self) -> str:
+        return str(self)
+
+
+MarkerType = Tuple[Union[Variable, Value], Op, Union[Variable, Value]]
+# MarkerAtom = Union[MarkerType, List["MarkerAtom"]]
```

Comment:
These names are a bit confusing. Perhaps use `MarkerAtom`, `MarkerItem`, `MarkerVar` and `MarkerList`, corresponding with their parse rules?